### DISCUSSION TITLE
What is the best way to use the library on MacOs ARM?

### DISCUSSION THREAD

**Discussion by Sing303** [ggml-org/whisper.cpp#2486] [Q&A]
What is the best way to use the library on MacOs ARM?

How to build, CoreML or Metal?
If Metal, how to build it correctly?

Which libraries should I use after building if I will use bindings in another programming language? (now there are a lot of files in the output, it's not clear which ones to use)

**Comment by Dev1an** [comment]
I tested on M1 Pro and M1 Max and for me the `GGML_USE_METAL` is faster than `WHISPER_USE_COREML`. From some basic inspection I noticed that the CoreML option was mostly offloading tasks to the Neural engine and the Metal option was offloading to the GPU. 
What is your experience?

**Reply by Sing303** [reply]
For some reason GGML_USE_METAL uses GPU only 15 - 20% and uses mostly CPU. I think maybe I built the executable file incorrectly
